scanpy/preprocessing/_simple.py

```diff
@@ -929,15 +931,65 @@ def scale_sparse(
         )
         X = X.toarray()
         copy = False  # Since the data has been copied
-    return scale_array(
-        X,
-        zero_center=zero_center,
-        copy=copy,
-        max_value=max_value,
-        return_mean_std=return_mean_std,
+        return scale_array(
+            X,
+            zero_center=zero_center,
+            copy=copy,
+            max_value=max_value,
+            return_mean_std=return_mean_std,
+            mask_obs=mask_obs,
+        )
+    elif isspmatrix_csc(X) and mask_obs is None:
+        return scale_array(
+            X,
+            zero_center=zero_center,
+            copy=copy,
+            max_value=max_value,
+            return_mean_std=return_mean_std,
+            mask_obs=mask_obs,
+        )
```

Comment:
No that doesn't hit the numba kernel. For CSC matrix I doesnt make sense to make sense to hit the numba kernel. If we use a map obs we need to transform into csr for the mask-obs subset for the `mean-var-calc`. I think what you suggest is fair that we also use the base implementation for none mask csr.

> blocks in numba code where the entire block is just excluded from the compiled code since it's a compile time constant.

I'll look into this and adjust the code based on this

Reply:
So I updated the kernel to use 2 `compile time constant`.